benchmarks/dataset_select_benchmark.py

```diff
@@ -0,0 +1,68 @@
+import conbench.runner
+import pyarrow.dataset
+
+from benchmarks import _benchmark
+
+
+@conbench.runner.register_benchmark
+class DatasetSelectBenchmark(_benchmark.Benchmark):
+    """
+    Read and filter a dataset on partition expressions.
+
+        DatasetSelectBenchmark().run(<source>, options...)
+
+    Parameters
+    ----------
+    source : str
+        A source name from the benchmarks source store.
+    cpu_count : int, optional
+        Set the number of threads to use in parallel operations (arrow).
+    iterations : int, default 1
+        Number of times to run the benchmark.
+    gc_collect : boolean, default True
+        Whether to do garbage collection before each benchmark run.
+    gc_disable : boolean, default True
+        Whether to do disable collection during each benchmark run.
+    run_id : str, optional
+        Group executions together with a run id.
+
+    Returns
+    -------
+    (result, output) : sequence
+        result : The benchmark result.
+        output : The output from the benchmarked function.
+    """
+
+    name = "dataset-select"
+    arguments = ["source"]
+    sources = ["nyctaxi_multi_parquet_s3_repartitioned"]
+    # This does not load a large amount of data in tests because we always pluck exactly one file from the dataset
```

Comment:
Minor: Please make this comment 2 lines long.

While not enforced, this repo follows the pep8 contention of 72 characters for documentation.

> For flowing long blocks of text with fewer structural restrictions (docstrings or comments), the line length should be limited to 72 characters.

-- https://www.python.org/dev/peps/pep-0008/